### DISCUSSION TITLE
Kit e Server Side Rendering

### DISCUSSION THREAD

**Discussion by icolumbro-asf** [italia/design-angular-kit#386] [General]
Stavo provando a partire con una nuova app in cui avrei voluto abilitare (soprattutto per questioni SEO) SSR ma mi sono dovuto subito fermare in quanto la libreria ha molti riferimenti all'oggetto globale window che sul server non è definito. C'è un modo per poter usare la libreria su un app con SSR?

**Comment by icolumbro-asf** [comment]
Ritorno su questo mi vecchio post, visto che avevo abbandonato la cosa ma ora il problema mi si sta ripresentando: se volessi realizzare un app Angular con supporto al Server-side rendering (SSR) principalmente per motivi di SEO, potrei utilizzare il kit o diventerebbe tutto troppo complicato in quanto i componenti del kit sono pensati principalmente per essere renderizzati lato client?

Spero di essermi spiegato, mi piacerebbe che mi aiutasti a chiarirmi le idee.

**Reply by riccardoperra** [reply]
@icolumbro-asf Da me per alcuni progetti avevamo fatto qualche prova diversi mesi fa, ma appena importavamo la libreria ottenevamo errori quali `window is not defined` o `document is not defined`. Credo quindi che l'SSR non sia supportata.

Abbiamo risolto utilizzando un design system interno con un tema che assomigliava a quello di boostrap Italia 😅 L'effort è stato decisamente minore ma avevamo già tutti componenti pronti all'uso
